docs/apis/subsystems/muc/index.md

```diff
@@ -0,0 +1,310 @@
+; staticaccelerationsize : [](int) This supplements the above setting by limiting the number of items in the caches persistent array of items.<br />Tweaking this setting lower will allow you to minimise the memory implications above while hopefully still managing to offset calls to the cache store.
+; ttl : [](int) A time to live for the data (in seconds). It is strongly recommended that you don't make use of this and instead try to create an event driven invalidation system (even if the event is just time expiring, better not to rely on ttl).<br />Not all cache stores will support this natively and there are undesired performance impacts if the cache store does not.
+; maxsize : [](int) If set this will be used as the maximum number of entries within the cache store for this definition.<br />Its important to note that cache stores don't actually have to acknowledge this setting or maintain it as a hard limit.
+; canuselocalstore : [](bool) This setting specifies whether the cache can safely be local to each frontend in a cluster which can avoid latency costs to a shared central cache server. The cache needs to be carefully written for this to be safe. It is conceptually similar to using $CFG->localcachedir (can be local) vs $CFG->cachedir (must be shared). Look at purify_html() in lib/weblib.php for an example.
+
+### Overriding a cache loader
+
+This is a super advanced feature and should not be done. ever. Unless you have an very good reason to do so.
+It allows you to create your own cache loader and have it be used instead of the default cache loader class. The cache object you get back from the make operations will be an instance of this class.
+; overrideclass : [](string) A class to use as the loader for this cache. This is an advanced setting and will allow the developer of the definition to take 100% control of the caching solution.<br />Any class used here must inherit the cache_loader interface and must extend default cache loader for the mode they are using.
+; overrideclassfile : [](string) Suplements the above setting indicated the file containing the class to be used. This file is included when required.
+
+### Specifying a data source
+
+This is a great wee feature, especially if your code is object orientated.
+
+It allows you to specify a class that must inherit the cache_data_source object and will be used to load any information requested from the cache that is not already being stored.
+
+When the requested key cannot be found in the cache the data source will be asked to load it. The data source will then return the information to the cache, the cache will store it, and it will then return it to the user as a request of their get request. Essentially no get request should ever fail if you have a data source specified.
+; datasource : [](string) A class to use as the data loader for this definition.<br />Any class used here must inherit the cache_data_source interface.
```

Comment:
As suggested with other similar cases before, this should be converted to a list too ;-) And [](string) should be replaced to [string] (otherwise, some of the jobs when you send the Pull-request will fail again).